### PR TITLE
Improve admin view

### PR DESCRIPTION
**Pull request by fadyyousry**
- readd collapse button

-  change author and category manager view

- remove show action from manager_book

-  change book edit&delete button 

- make concern for megamenu

- remove DRY in localization

### CODE REVIEW

File: app/controllers/concerns/search_menus.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+module SearchMenus
```
Comment (Magdyx) [inline app/controllers/concerns/search_menus.rb:1]: I don;t think this is a good name. Maybe MenuDataPreload or somthing like that

File: app/controllers/concerns/search_menus.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    included do
+        before_action :authors_index, :categories_index
+    end
+
+    def authors_index
+        @authors = Author.all
+    end
+
+    def categories_index
+        @categories =Category.all
```
Comment (Magdyx) [inline app/controllers/concerns/search_menus.rb:12]: space

File: app/javascript/admin/table_with_details.js [hunk 3]
```diff
@@ -36,22 +35,6 @@ $(document).ready(function() {
...
-                    "visible": false
-                },
-                {
-                    "targets": [ 3 ],
-                    "visible": false
-                },
-                {
-                    "targets": [ 4 ],
-                    "visible": false
-                }
```
Comment (Magdyx) [inline app/javascript/admin/table_with_details.js:54]: what are those and why they are no longer needed?

File: app/views/manager/authors/index.html.erb [hunk 4]
```diff
@@ -1,33 +1,38 @@
...
+          <div class="row">
+            <% author.books.each do |book| %>
+                <div class="col-sm-6 col-md-2">
+                  <div class="thumbnail">
+                    <%= image_tag book.image_link, class: "card-img-top" %>
+                    <div class="caption">
+                      <h5><%= book.title %></h5>
+                    </div>
+                  </div>
+                </div>
```
Comment (Magdyx) [inline app/views/manager/authors/index.html.erb:30]: is this part the same as in the book index? can it be converted to a partial and just pass the book to it?

File: app/views/manager/books/index.html.erb [hunk 5]
```diff
@@ -32,9 +23,18 @@
...
+      <td>
+        <div class="center">
+          <%= link_to edit_manager_book_path(book.id) do %>
+            <i class="fa fa-pencil" aria-hidden="true"></i>
+          <% end %>
+          <%= link_to manager_book_path(book.id), method: :delete do %>
+            <i class="fa fa-times" aria-hidden="true"></i>
+          <% end %>
+        </div>
+      </td>
```
Comment (Magdyx) [inline app/views/manager/books/index.html.erb:35]: I think this part is made as a partial that you can use in users, did you check it?

File: app/controllers/concerns/search_menus.rb (hunk 1 above)
Comment (fadyyousry) [reply app/controllers/concerns/search_menus.rb:1]: done

File: app/controllers/concerns/search_menus.rb (hunk 2 above)
Comment (fadyyousry) [reply app/controllers/concerns/search_menus.rb:12]: done

File: app/javascript/packs/admin.js [hunk 6]
```diff
@@ -11,8 +11,8 @@ require('magnific-popup')
 require('datatables.net-bs')
 require('bootstrap-tagsinput')
 // files
-//require('admin/theme')
+require('admin/theme')
 require('admin/sidebar')
 require('admin/add_book')
-require('admin/datatable_book')
 require('admin/users_table')
+require('admin/table_with_details')
```
Comment (Magdyx) [inline app/javascript/packs/admin.js:18]: I don't think this is a good name as well. `books_table` is fine

File: app/javascript/admin/table_with_details.js (hunk 3 above)
Comment (fadyyousry) [reply app/javascript/admin/table_with_details.js:54]: Useless code I forgot to delete it last time.
It makes columns hidden, then used their data to fill the extended part of table. anyway, I used another approach.

File: app/views/manager/authors/index.html.erb (hunk 4 above)
Comment (fadyyousry) [reply app/views/manager/authors/index.html.erb:30]: No, it isn't

File: app/views/manager/books/index.html.erb (hunk 5 above)
Comment (fadyyousry) [reply app/views/manager/books/index.html.erb:35]: do you mean this partial 
image

you use edit_manager_user_path & manager_user_path. it isn't changeable

File: app/javascript/packs/admin.js (hunk 6 above)
Comment (fadyyousry) [reply app/javascript/packs/admin.js:18]: it isn't for book only it's for author and category too

File: app/views/manager/authors/index.html.erb (hunk 4 above)
Comment (fadyyousry) [reply app/views/manager/authors/index.html.erb:30]: done

File: app/views/manager/books/index.html.erb [hunk 7]
```diff
@@ -1,12 +1,6 @@
-<% title t('activerecord.models.book.other') %>
```
Comment (Magdyx) [inline app/views/manager/books/index.html.erb:1]: what!

File: app/views/manager/books/index.html.erb (hunk 5 above)
Comment (Magdyx) [reply app/views/manager/books/index.html.erb:35]: update it with ruby's send method and use it

File: app/views/shared/_navbar.html.erb [hunk 8]
```diff
@@ -19,14 +19,14 @@
         <%= link_to t('default.shop'), books_path, class: "nav-link" %>
       </li>
       <li id="authors_list" class="navbar-item dropdown has-megamenu">
-        <%= link_to t('default.author.name.other'), '#',
-         class: "nav-link dropdown-toggle", :"data-toggle" => "dropdown"%>
-         <%= render 'shared/megamenu', list: Author.all %>
+        <%= link_to t('activerecord.models.author.other'), '#',
+         class: "nav-link dropdown-toggle", "data-toggle":"dropdown"%>
```
Comment (Magdyx) [inline app/views/shared/_navbar.html.erb:23]: why are you changing this line?

File: app/views/manager/books/index.html.erb (hunk 7 above)
Comment (fadyyousry) [reply app/views/manager/books/index.html.erb:1]: merge conflict

Comment (Magdyx) [reply app/views/manager/books/index.html.erb:1]: this line shouldn't be removed

File: app/views/shared/_navbar.html.erb (hunk 8 above)
Comment (Magdyx) [reply app/views/shared/_navbar.html.erb]: would you mind changing it back

**Pull request closed without merging**